Floe 112 - Styling for the demo

jhung opened the pull request:
Added styling and structure to match the Metadata demo design.

cindyli committed:
FLOE-113: Implemented the audio metadata panel.

cindyli committed:
FLOE-113: Replaced tabs in style.css with spaces.

cindyli committed:
FLOE-113: Fixed the panel flickering issue in the audio metadata panel when switching btw audio attributes.

michelled committed:
Merge remote-tracking branch 'cindyli/FLOE-113'

* cindyli/FLOE-113:
  FLOE-113: Fixed the panel flickering issue in the audio metadata panel when switching btw audio attributes.
  FLOE-113: Replaced tabs in style.css with spaces.
  FLOE-113: Implemented the audio metadata panel.

michelled commented on demos/css/simpleEditor.css:
```diff
@@ -0,0 +1,172 @@
+}
+
+/* Bootstrap small width */
+@media (min-width: 768px) {
```
We should probably use em based media queries.

jhung replied:
The pixel values (768px etc.) is from bootstrap's media query breakpoints. So our media queries are set to match the same breakpoints. Otherwise our styles may apply at different points which will look confusing.

Regarding em vs pixels, since these media queries are for client devices - it makes sense to have pixels since we're dealing with physical sizes and nor relative sizes.

This doesn't change the fact that bootstrap uses pixels for every measurement - but this is customizable at least by customizing the bootstrap build.

michelled commented on demos/js/simpleEditor.js:
```diff
@@ -189,4 +214,120 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+
+    fluid.simpleEditor.insertVideo.reset = function (that) {
+        that.setModel({});
+        that.updateActiveState();
```
shouldn't this happen as a result of the model changing? ie. based on the modelChanged event?

jobara replied:
It is necessary to update the state here as a reset may be called before a model value is saved, but the text has been entered into the url field. I've left a comment next to line 329 indicating that it is clearing the model.

jobara committed:
Merge branch 'master' into FLOE-112

jobara committed:
Merge remote-tracking branch 'jhung/FLOE-112' into FLOE-112

jobara committed:
Merge remote-tracking branch 'jhung/FLOE-112' into FLOE-112

jobara committed:
Merge remote-tracking branch 'jhung/FLOE-112' into FLOE-112

jobara committed:
FLOE-112: Preventing default action for OK button

Preventing the default action (form submission) for the OK button used to enter a video url. Also added a comment to the reset method.

michelled commented:
This looks good - go ahead and push it up.
